test/minimallist/generator_test.cljc

```diff
@@ -385,9 +385,23 @@
   (let [model (-> (h/map [:a fn-int?])
                   (h/with-optional-entries [:b fn-string?]))
         sample (tcg/sample (gen model) 100)]
-    (is (and (every? (partial valid? model) sample)
-             (some (fn [element] (contains? element :b)) sample)
-             (some (fn [element] (not (contains? element :b))) sample))))
+    (is (every? (partial valid? model) sample))
+    (is (every? (fn [element] (contains? element :a)) sample))
+    (is (some (fn [element] (contains? element :b)) sample))
+    (is (some (fn [element] (not (contains? element :b))) sample))
+    (is (some (fn [element] (and (contains? element :a)
+                                 (not (contains? element :b)))) sample)))
+
+  (let [model (-> (h/map)
+                  (h/with-optional-entries
+                    [:str (-> (h/fn string?)
+                              (h/with-test-check-gen tcg/string-ascii))])
+                  (h/with-entries [:int (-> (h/fn int?)
+                                            (h/with-test-check-gen tcg/nat))]))
+        sample (tcg/sample (gen model) 20)]
+    (is (some (fn [element] (and (contains? element :int)
+                                 (not (contains? element :str))))
+              sample)))
```

Comment:
This use case is already covered above (L385~386), this can be removed.

Reply:
Hold a second ...

I noticed that the test at lines 385~386 does not trigger the bug. The reason the bug is triggered in your test is related to the order of the keys.

This model will trigger the bug, for example:
```
  (let [model (-> (h/map)
                  (h/with-optional-entries [:b fn-string?])
                  (h/with-entries [:a fn-int?]))
        sample (tcg/sample (gen model) 100)]
    (is (and (every? (partial valid? model) sample)
             (every? (fn [element] (contains? element :a)) sample)
             (some (fn [element] (contains? element :b)) sample)
             (some (fn [element] (not (contains? element :b))) sample))))
```

I suggest to replace your test by this snippets instead.

Reply:
**Update**: I propose to replace the existing test by the following and remove the additional test.

```
  (let [model (-> (h/map [:a fn-int?])
                  (h/with-optional-entries [:b fn-string?])
                  (h/with-entries [:c fn-int?])
                  (h/with-optional-entries [:d fn-string?]))
        sample (tcg/sample (gen model) 100)]
    (is (and (every? (partial valid? model) sample)
             (every? (fn [element] (contains? element :a)) sample)
             (some (fn [element] (contains? element :b)) sample)
             (some (fn [element] (not (contains? element :b))) sample)
             (every? (fn [element] (contains? element :c)) sample)
             (some (fn [element] (contains? element :d)) sample)
             (some (fn [element] (not (contains? element :d))) sample))))
```